### PR TITLE
Allowing manual 'sync now' after a reasonable time in enqueued state

### PR DESCRIPTION
**Pull request by Kartones**
Fixes https://github.com/CartoDB/cartodb-platform/issues/639

### CODE REVIEW

**Comment by Kartones** [comment]
@rafatower CR of the small Ruby change. Just allows to also sync if at `queued` state for too long (probably stuck, as hasn't been processed). Is a last resort "feature" but nice to have.

No tests but checked manually

@xavijam CR of the small JS change (back when I added queued state I didn't add it to the JS states list, sorry)

**Comment by iriberri** [comment]
Oh, super interesting. Thanks!

**Comment by xavijam** [comment]
No problem in JS side.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4497/))

File: app/models/synchronization/member.rb [hunk 1]
```diff
@@ -131,7 +131,11 @@ def enqueue
 
       # @return bool
       def can_manually_sync?
-        (self.state == STATE_SUCCESS || self.state == STATE_FAILURE) && (self.ran_at + SYNC_NOW_TIMESPAN < Time.now)
+        # Last sync ok, last sync failed, or too much time in queued state
+        (  self.state == STATE_SUCCESS || 
+           self.state == STATE_FAILURE || 
+          (self.state == STATE_QUEUED && self.updated_at + SYNC_NOW_TIMESPAN < Time.now)
```
Comment (rafatower) [inline app/models/synchronization/member.rb]: I'd just reverse the time comparison for clarity:
```
Time.now > (self.updated_at + SYNC_NOW_TIMESPAN)
```
but that's a matter of taste I guess.

**Comment by rafatower** [comment]
:+1:

**Pull request merged**